pyirf/scripts/make_DL3.py

```diff
@@ -0,0 +1,338 @@
+    )
+
+    parser.add_argument(
+        "--obs_time",
+        type=str,
+        required=True,
+        help="An observation time written as (value.unit), e.g. '50.h'",
+    )
+
+    parser.add_argument(
+        "--pipeline",
+        type=str,
+        required=True,
+        help="Name of the pipeline that has produced the DL2 files.",
+    )
+
+    parser.add_argument(
+        "--debug", action="store_true", help="Print debugging information."
+    )
+
+    args = parser.parse_args()
+
+    # INPUT FROM THE CONFIGURATION FILE
+
+    cfg = load_config(args.config_file)
+
+    # Add obs. time to the configuration file
+    str_obs_time = args.obs_time.split(".")
+    cfg["analysis"]["obs_time"] = {
+        "value": float(str_obs_time[0]),
+        "unit": str(str_obs_time[-1]),
+    }
+
+    # Get input directory
+    indir = cfg["general"]["indir"]
+
+    # Get template of the input file(s)
+    template_input_file = cfg["general"]["template_input_file"]
+
+    # Get output directory
+    outdir = os.path.join(
+        cfg["general"]["outdir"],
+        "irf_{}_Time{}{}".format(  # name could change in future
+            args.pipeline,
+            # cfg["analysis"]["thsq_opt"]["type"], # not needed now
+            # cfg["analysis"]["obs_time"]["value"],
+            # cfg["analysis"]["obs_time"]["unit"],
+            args.obs_time.split(".")[0],
+            args.obs_time.split(".")[-1],
+        ),
+    )  # and create it if necessary
+    if not os.path.exists(outdir):
+        os.makedirs(outdir)
+
+    # =========================================================================
+    #                   READ DL2 DATA AND STORE IT ACCORDING TO GADF
+    # =========================================================================
+
+    # Load FITS data
+    particles = ["gamma", "electron", "proton"]
+    evt_dict = dict()  # Contain DL2 file for each type of particle
+    for particle in particles:
+        if args.debug:
+            print(f"Loading {particle} DL2 data...")
+        infile = os.path.join(indir, template_input_file.format(particle))
+        evt_dict[particle] = read_FITS(config=cfg, infile=infile)
+
+    # =========================================================================
+    #               PRELIMINARY OPERATIONS FOR SPECIFIC PIPELINES
+    # =========================================================================
+
+    # Some pipelines could provide some of the DL2 data in different ways
+    # After this part, DL2 data is supposed to be equivalent, regardless
+    # of the original pipeline.
+
+    # Later we should move this out of here, perhaps under a "utils" module.
+
+    if args.pipeline == "EventDisplay":
+
+        # THETA IS SUPPOSED TO BE IN DL2 DATA ACCORDING TO GADF
+        # EventDisplay provides true and reconstructed directions, so we
+        # calculate it here and we add it to the tables.
+
+        # WARNING:
```

Comment:
Then please directly implement it correctly, what is actually needed here? Also for mono telescopes we also have wobble observations. So there are simulations where we simulated a point source that is not at the center of the field of view.

Reply:
We should just always make the distinction and call the variables accordingly.

Reply:
not sure if we need to modify it...

THETA = angular_separation(
                evt_dict[particle]["TRUE_AZ"],
                evt_dict[particle]["TRUE_ALT"],
                evt_dict[particle]["AZ"],
                evt_dict[particle]["ALT"],
            )

takes already into account off-axis events, am I right?
But in that case, theta 
- is from the FoV center, so the name would be the wrong one,
- I guess TRUE_AZ and TRUE_ALT should be modified in DETX and DETY (I guess they are already in the sky)
But this info is not in the EventDisplay DL2 data

Though, in general, I would leave any off-axis-related modification to a dedicated PR - we are trying to fix the on-axis for the moment.

Reply:
I don't understand what you are trying to say with your second point. DETX and DETY are something completely different.

Reply:
ah you mean the warning part?